cmd/create/create_agent.go

```diff
@@ -0,0 +1,439 @@
+		agentName = requestedContext
+	}
+
+	o.Name = agentName
+
+	// set namespace
+	promptSetNamespace := promptui.Prompt{
+		Label:  fmt.Sprintf("Provide a namespace where the agent will be installed [default=%s]", defaultNamespaceName),
+		Stdout: &util.BellSkipper{},
+	}
+
+	namespaceName, err := promptSetNamespace.Run()
+	if err != nil {
+		return errors.New(fmt.Sprintf("failed to set the namespace; %v\n", err))
+	}
+
+	if lo.IsEmpty(namespaceName) {
+		namespaceName = defaultNamespaceName
+	}
+
+	o.Namespace = namespaceName
+
+	// fetch the list of credentials
+	existingCredentials, err := o.ArmoryClient.Credentials().List(ctx)
+	if err != nil {
+		return err
+	}
+
+	// create new credentials
+	credentials := o.createCredentials()
+	existingCredential, credentialsExists := lo.Find(existingCredentials, func(c *model.Credential) bool {
+		return credentials.Name == c.Name
+	})
+
+	if credentialsExists {
+		// recreate credentials
+		promptRecreateCredentials := promptui.Prompt{
+			Label:     fmt.Sprintf("A Client Credential named %s already exists. Do you want to generate a new Client Credentials?", o.Name),
+			IsConfirm: true,
+			Default:   "Y",
+			Stdout:    &util.BellSkipper{},
+		}
+
+		if _, err := promptRecreateCredentials.Run(); err != nil {
+			return errors.New(fmt.Sprintf("Exiting %s\n", err))
+		}
+
+		err = o.ArmoryClient.Credentials().Delete(ctx, existingCredential)
+		if err != nil {
+			return errors.New(fmt.Sprintf("failed to delete credentials; %v\n", err))
+		}
+	}
+
+	credentials, err = o.ArmoryClient.Credentials().Create(ctx, credentials)
+	if err != nil {
+		return err
+	}
+
+	existingRoles, err := o.ArmoryClient.Roles().ListForMachinePrincipals(ctx, o.configuration.GetCustomerEnvironmentId())
+	if err != nil {
+		return err
+	}
+
+	// add the RNA role to the newly created credentials
+	rol, rolExists := lo.Find(existingRoles, func(c model.RoleConfig) bool {
+		return "Remote Network Agent" == c.Name
+	})
+
+	if !rolExists {
+		return errors.New("The default role Remote Network Agent role was missing, please ask your tenant admins to recreate it.")
+	}
+
+	_, err = o.ArmoryClient.Credentials().AddRoles(ctx, credentials, []string{rol.ID})
+	if err != nil {
+		return err
+	}
+
+	o.credentials = credentials
+
+	// create new namespace if not exist
+	if exist, _ := o.namespaceExist(); !exist {
+		_, err := o.createNamespace()
+		if err != nil {
+			return errors.New(fmt.Sprintf("failed to create namespace; %v\n", err))
+		}
+	}
+
+	// create new secret
+	createSecretOptions := metav1.CreateOptions{}
+	secret := o.createSecret()
+	secret, err = o.KubernetesClient.Secrets(o.Namespace).Create(ctx, secret, createSecretOptions)
+	if err != nil {
+		return errors.New(fmt.Sprintf("failed to create secret; %v\n", err))
+	}
+
+	// apply manifests
+	err = o.apply(o.Namespace, fmt.Sprintf("%s/kubernetes/agent/manifest?agentIdentifier=%s&namespace=%s", o.configuration.GetArmoryCloudAddr(), o.Name, o.Namespace))
```

Comment:
This makes me really sad because its missing all of the configuration options that the helm chart exposes and we are not doing a great job of managing this and I really am pushing for deleting this endpoint because its a maintenance nightmare.

Reply:
Some configuration options that this doesn't expose that I think would be needed if this is going to be a supported method of RNA installation:

**Proxy Configuration**
https://github.com/armory-io/remote-network-agent-helm-chart/blob/6ae0c71b77377647d1498f902da6c724a3aa453c/values.yaml#L51-L70

**IAM**
https://github.com/armory-io/remote-network-agent-helm-chart/blob/6ae0c71b77377647d1498f902da6c724a3aa453c/values.yaml#L80-L84

**Logging**
https://github.com/armory-io/remote-network-agent-helm-chart/blob/6ae0c71b77377647d1498f902da6c724a3aa453c/values.yaml#L100-L106

**Environment**
https://github.com/armory-io/remote-network-agent-helm-chart/blob/6ae0c71b77377647d1498f902da6c724a3aa453c/templates/deployment.yaml#L58
:point_up: with this we could use the CLI ourselves to install agents in kind clusters to point to dev and staging